File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int getAttack(int number, int limit, int power) {
        int cnt = 0;
        int i = 1;
        while(i*i < number) {
            if(number % i == 0) {
                cnt += 2;
            }
            i++;
        }
        if(i*i == number) {
            cnt += 1;
        }

        return cnt > limit ? power : cnt;
    }
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i = 0; i<number; i++){
            if(i == 0) answer+= 1;
            else {
                answer += getAttack(i+1, limit, power);
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.kt
import kotlin.math.sqrt

class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer = 0
        fun getPower(p: Int) : Int = if(p > limit) power else p

        (1..number).forEach { n ->
            if(n == 1) {
                answer += getPower(n)
                return@forEach
            } else {
                val sqrt = sqrt(n.toDouble()).toInt()
                var cnt = 0
                (1..sqrt).forEach { s ->
                    if(n%s == 0) {
                        cnt += 2
                    }

                }
                val p = if(sqrt*sqrt == n) cnt-1 else cnt
                answer += getPower(p)
            }
        }
        return answer
    }
}